apps/sms/views/shared/js/activity_handler.js

```diff
@@ -114,43 +117,51 @@ var ActivityHandler = {
           return size;
         }, 0);
 
-        if (size > Settings.mmsSizeLimitation) {
-          Utils.alert({
-            id: 'attached-files-too-large',
-            args: {
-              n: activityData.blobs.length,
-              mmsSize: (Settings.mmsSizeLimitation / 1024).toFixed(0)
-            }
-          }).then(() => ActivityClient.postResult());
-          return;
-        }
+        dataPromise = MozSettingsClient.mmsSizeLimitation().then((limit) => {
```

Comment:
My head is emptied after just a few days I review patch, so I can miss something now, but handling all errors in one place sounds good to me if not  too difficult :)